rcl_yaml_param_parser/src/parser.c

```diff
@@ -1105,6 +1105,10 @@ static void * get_value(
   {
     errno = 0;
     endptr = NULL;
+    if (strcasecmp(value, ".nan") == 0) {
+      char * tmp_value = (char *)value;
+      snprintf(tmp_value, strlen(tmp_value), "%s", "nan");
+    }
```

Comment:
While this works, it breaks the contract of the function.  That is, value is passed in as `const char * const value`, and so the caller should (rightly) assume that the string won't be modified.

I think in order to do what you want to do here, we need to advance the value pointer by one byte so it points to the `n` at the front.

Reply:
> While this works, it breaks the contract of the function. That is, value is passed in as const char * const value, and so the caller should (rightly) assume that the string won't be modified.

That make sense, but i am not sure whether  ~~const~~ char is really neccessary.

> I think in order to do what you want to do here, we need to advance the value pointer by one byte so it points to the n at the front.

Sorry, i dont't get it. The `value` pointer is decorated by const, so it's a read-only pointer, how can i change its direction ?

Reply:
> Sorry, i dont't get it. The `value` pointer is decorated by const, so it's a read-only pointer, how can i change its direction ?

We'll have to be a little creative :).

Thinking about it more, I think the best thing to do here is probably to make a copy of the original data, then try to parse that as a double.  That way we keep the current function contract, but can also move as we want.

When you write tests for this, please also consider the case where the user is passing in `nananana` as a string; we'll want to make sure we don't improperly consider that a double.

Also, while we are here, I think we should probably add in support for YAMLs "Inf", just to be complete.